### DISCUSSION TITLE
KeyExists assert

### DISCUSSION THREAD

**Discussion by qwerty2586** [lainio/err2#9] [Ideas]
Sometime I struggle with checking if map has some key i need, for example from unmarshaled json. 

Something like this would be nice
```go
func (asserter Asserter) KeyExists[T comparable, V any](m map[T]V, key T, a ...any) {
	_, exists := m[key]
        if !exists {
               defMsg := fmt.Sprintf("key %v doesnt exist", key)
               asserter.reportAssertionFault(defMsg, a...)
        }
}
```
Not sure if this implementation is ok

**Comment by lainio** [comment]
That's an excellent idea. I'll add it to my backlog. ETA in the next release, hopefully in a week.

**Comment by lainio** [comment]
Hi @qwerty2586, the latest version, v0.8.10, has `assert.MKeyExists`. The `M` prefix is there for the future -- mainly. And because it follows the current naming convention: `S` = slice, `M` = map, `C` = channel, `I` interface.

I have migrated one of our large code bases to use `err2` instead of `testfify`, and the first feelings have been great.

Please let me know if you find any bugs or anything.

**Reply by qwerty2586** [reply]
Looks great. Your lib will soon eradicate all these pesky one-time variables.
